### DISCUSSION TITLE
Is `ats_percel` already included in `Gwtadv`?

### DISCUSSION THREAD

**Discussion by mbakker7** [modflowpy/flopy#2347] [General]
I want to use the new `ats_percel` option in `Gwtadv` to adjust the time step based on a Courant number. 
This seems included in the latest MODFLOW 6 nightly built, but not yet in Flopy 3.9.0.dev2. 
Am I missing something or is Flopy not yet updated?

**Comment by christianlangevin** [comment]
Take a look at the [generate_classes](https://github.com/modflowpy/flopy/blob/develop/.docs/md/generate_classes.md) utility in flopy.  This should get you up and running.  The following command will update flopy to support the development version of MODFLOW 6.

```
python -m flopy.mf6.utils.generate_classes --ref develop
```

**Comment by mbakker7** [comment]
Thanks. Will try that. Follow up question:
The MODFLOW manual now says: "the advective fractional cell distance (ATS_PERCEL)” is not “exceeded” "for any active cell in the grid”. That is nice, but seems maybe too rigorous? When the upstream concentration is (almost) the same as the concentration in the downstream cell, there is no need to reduce the time step. For example, when injecting a fluid with a certain concentration, the cells around the well will all have similar concentration a short time after injection started, and there is no need to take those cells into account to reduce the time step.  Is something like that done? Or is it a simple check of the Courant number regardless of concentrations?

**Reply by mbakker7** [reply]
Small additional question: What does MT3D do? That has an ATS_PERCEL parameter. Does that check upstream concentrations or is it also "just" a Courant number check?

**Reply by christianlangevin** [reply]
The approach implemented in MODFLOW 6 is close to what is done in MT3D/SEAWAT, except it is generalized for unstructured grids.  It is essentially just a courant check. It does not look at adjacent cell concentrations or do anything fancy.  Perhaps more could be done here, but I'm not sure it straightforward to come up with a more efficient, but general, filtering of areas where the constraint should be applied.
